### PR TITLE
[SHLWAPI] 1/3-implement SHAutoComplete

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Retrial of #3214. Try to implment `shlwapi!SHAutoComplete` function.

JIRA issue: [CORE-9281](https://jira.reactos.org/browse/CORE-9281)

## Proposed changes

- Modify `CMakeLists.txt` to add and compile `autocomp.cpp`.
- Add some functions to implement `SHAutoComplete`.
- Add `shlwapi!SHPinDllOfCLSID` function prototype into `shlwapi_undoc.h`.
- Try to implement `SHAutoComplete`.

### CODE REVIEW

**Comment by katahiromz** [comment]
`CLSID_ACListISF` is lacking to work.

File: dll/win32/shlwapi/autocomp.cpp [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    }
+
+    // for all the MRU items
+    CStringW strData;
+    for (INT i = 0; i <= L'z' - L'a' && i < strMRUList.GetLength(); ++i)
+    {
+        // Build a registry value name
+        WCHAR szName[2];
+        szName[0] = strMRUList[i];
+        szName[1] = 0;
```
Comment (learn-more) [inline dll/win32/shlwapi/autocomp.cpp:77]:
```suggestion
        WCHAR szName[2] = {strMRUList[i]};
```

Comment (katahiromz) [reply dll/win32/shlwapi/autocomp.cpp]: OK.

**Comment by katahiromz** [comment]
Give me 7 days. I have to investigate the target COM objects deeply...

**Comment by katahiromz** [comment]
Ready to go!

**Pull request merged**